Automattic/jetpack
# Newsletter categories: Render non-interactive newsletter category pills in the editor

Closes https://github.com/Automattic/wp-calypso/issues/80904

## Proposed changes:
- This adds the newsletter categories as pills in the editor.

<img width="682" alt="CleanShot 2023-08-24 at 16 23 51@2x" src="https://github.com/Automattic/jetpack/assets/528287/c91c2dc3-3bc3-400f-a35c-220c89c36bed">

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:
For completeness, please test this on WPCOM & Jetpack self hosted.

1. Checkout this PR and run the command below for WPCOM or build it and run it though JT
2. Mark some categories as newsletter categories: `POST /wpcom/v2/sites/<siteid>/newsletter-categories/<categoryid>` 
3. Create a new post/page & add the subscribe block
4. The newsletter categories should be above the input field
